src/Microsoft.DotNet.Core.Build.Tasks/GenerateRuntimeConfigurationFiles.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.DotNet.ProjectModel;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NuGet.ProjectModel;
+
+namespace Microsoft.DotNet.Cli.Tasks
+{
+    /// <summary>
+    /// Generates the $(project).runtimeconfig.json and optionally $(project).runtimeconfig.dev.json files
+    /// for a project.
+    /// </summary>
+    public class GenerateRuntimeConfigurationFiles : Task
+    {
+        [Required]
+        public string RuntimeOutputPath { get; set; }
+
+        [Required]
+        public string OutputName { get; set; }
+
+        [Required]
+        public string LockFilePath { get; set; }
+
+        public string RawRuntimeOptions { get; set; }
+
+        public bool IncludeDevConfig { get; set; }
+
+        private LockFile LockFile { get; set; }
+
+        public override bool Execute()
+        {
+            LockFile = LockFileCache.Instance.GetLockFile(LockFilePath);
```

Comment:
I tend to see Singletons as not great constructs. They make the code very hard to test and change. And going with my previous comment. We should be adding test coverage to this and other tasks that we implement.